src/pip/_internal/req/req_install.py

```diff
@@ -372,7 +367,16 @@ def ensure_build_location(self, build_dir):
         if not os.path.exists(build_dir):
             logger.debug('Creating directory %s', build_dir)
             os.makedirs(build_dir)
-        return os.path.join(build_dir, name)
+        actual_build_dir = os.path.join(build_dir, name)
+        # If autodeleting we actually want to respect the globally-configured
+        # deletion option, which we indicate with `None`.
```

Comment:
This doesn't read very clearly to me -- would it be OK to rephrase this to something like:

```
`None` indicates that we respect the globally-configured deletion settings,
which is what we actually want when auto-deleting.
```

Reply:
Sure, done [here](https://github.com/pypa/pip/pull/7696/files#diff-47a4ece629e4698a5eaefea075d3cbdfR371-R372).